lib/nats.js

```diff
@@ -797,6 +801,23 @@ Client.prototype.processInbound = function() {
 	if (client.checkTLSMismatch() === true) {
 	  return;
 	}
+
+  // Always try to read the connect_urls from info
+  if(client.info.connect_urls && client.info.connect_urls.length > 0) {
```

Comment:
Even with this link, I am sorry but it looks horrible. It is very difficult to know what is in which block.